pkg/apis/admissionregistration/types.go

```diff
@@ -115,6 +115,223 @@ const (
 
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
 
+// ValidatingAdmissionPolicy describes the definition of an admission validation policy that accepts or rejects and object without changing it.
+type ValidatingAdmissionPolicy struct {
+	metav1.TypeMeta
+	// Standard object metadata; More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata.
+	// +optional
+	metav1.ObjectMeta
+	// Specification of the desired behavior of the AdmissionPolicy.
+	Spec ValidatingAdmissionPolicySpec
+}
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ValidatingAdmissionPolicyList is a list of ValidatingAdmissionPolicy.
+type ValidatingAdmissionPolicyList struct {
+	metav1.TypeMeta
+	// Standard list metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#types-kinds
+	// +optional
+	metav1.ListMeta
+	// List of ValidatingAdmissionPolicy.
+	Items []ValidatingAdmissionPolicy
+}
+
+// ValidatingAdmissionPolicySpec is the specification of the desired behavior of the AdmissionPolicy.
+type ValidatingAdmissionPolicySpec struct {
+	// ParamSource specifies the kind of resources used to parameterize this policy which has to be a CRD.
+	// This is set to optional because we might support the singleton binding use case in the future.
+	// +optional
+	ParamSource *ParamSource
```

Comment:
Question: Should GroupVersionKind be used here instead of ParamSource?
https://github.com/kubernetes/kubernetes/blob/afbce897a9c7956fd3219b64abacf46fa5d3e6aa/staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go#L142-L146

Reply:
I'm okay with the type being an existing type. I do think we should go with `apiVersion` and `kind`